src/hotspot/share/asm/register.hpp

```diff
@@ -63,7 +63,7 @@ enum { name##_##type##EnumValue = (value) }
 inline constexpr type as_ ## type(int encoding) {                       \
   return impl_type::first() + encoding;                                 \
 }                                                                       \
-extern impl_type all_ ## type ## s[reg_count + 1] INTERNAL_VISIBILITY;  \
+INTERNAL_VISIBILITY extern impl_type all_ ## type ## s[reg_count + 1];  \
```

Comment:
I'm surprised this works for gcc-windows with the existing INTERNAL_VISIBILITY definition.  I guess all uses
of REGISTER_IMPL_DECLARATION are in the context of the objectionable _WIN32 undef/redef dances.